File: FlareCore/Logger/LogVerbosity.cs
// CopyRight FlareHorz Engine Development Team. All Rights Reserved.

namespace FlareCore.Logger;

/// <summary>
/// Verbosity of the log.
/// </summary>
[FlareCoreAPI]
public enum ELogVerbosity
{
	/// <summary>
	/// Not used.
	/// </summary>
	NoLogging = 0,
	
	/// <summary>
	/// Prints an error to console (and log file).
	/// </summary>
	Error = 1,
	
	/// <summary>
	/// Prints a warning to console (and log file).
	/// </summary>
	Warning = 2,
	
	/// <summary>
	/// Prints a message to console (and log file).
	/// </summary>
	Info = 3,
	
	/// <summary>
	/// Prints a debug message to console (and log file).
	/// Works only in Debug configuration.
	/// </summary>
	Debug = 4,
	
	/// <summary>
	/// Enable all logging verbosities.
	/// </summary>
	All = 5
}

File: Source/Runtime/Core/Private/Logging/LogVerbosity.cpp
// CopyRight FlareHorz Team. All Rights Reserved.


#include "Logging/LogVerbosity.h"

const TCHAR* LogVerbosityToString(const ELogVerbosity Verbosity)
{
	switch (Verbosity)
	{
		case ELogVerbosity::Fatal:			return FH_TEXT("Fatal");
		case ELogVerbosity::Error:			return FH_TEXT("Error");
		case ELogVerbosity::Warning:		return FH_TEXT("Warning");
		case ELogVerbosity::Log:			return FH_TEXT("Log");
		case ELogVerbosity::Verbose:		return FH_TEXT("Verbose");
		case ELogVerbosity::VeryVerbose:	return FH_TEXT("VeryVerbose");
		default:							return FH_TEXT("Unknown");
	}
}

